File: 209-MinimumSizeSubarraySum.py
class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        sm = 0
        i = 0
        min_length = float('inf')
        for j in range(len(nums)):
            sm += nums[j]
            while sm >= target:
                min_length = min(min_length, j - i + 1)
                sm -= nums[i]
                i += 1
        
        return min_length if min_length != float('inf') else 0
                
        
        

File: 209-MinimumSizesubarraySum.cpp
class Solution {
public:
    int minSubArrayLen(int target, vector<int>& nums) {
        int min_length = INT_MAX;
        int sum = 0;
        int i = 0;
        
        for (int j = 0; j < nums.size(); ++j) {
            sum += nums[j];
            while (sum >= target) {
                min_length = min(min_length, j - i + 1);
                sum -= nums[i++];
            }
        }
        
        return min_length != INT_MAX ? min_length : 0;
    }
};